### DISCUSSION TITLE
Set MTU for all containers

### DISCUSSION THREAD

**Discussion by pongraczi** [nextcloud/all-in-one#3051] [Questions] [answered]
Hi,

I would like to set the MTU for all the containers to 1400.
Unfortunately it is not trivial and seems not reboot proof.

What is the correct way to set the MTU for all interfaces which will survive a reboot?

I tried the following:  
* edit /etc/docker/daemon.json and add "mtu": 1400
* I created a compose yaml and I set the default mtu to 1400, this didn't work as expected
* I set mtu on the host, using this command: **ip link set group default mtu 1400** , which succeed, only backdraw, it needs to run "manually" 

Is there any way to set mtu permanently to all the containers?

Background info:  
* I have servers at a provider, let's call it *noname* (not relevant)
* I installed ProxmoxVE and I created an lxc, which is not on the public network (vmbr0) but in the local lan (vmbr1)
* for technical reasons, I need to use **mtu=1400** on the local network
* that lxc is pretty fine running docker images without problem
* I started aio on this lxc and for first, it was fine (more or less, for first time I run into the mtu mismatch, but somehow I managed to change the mtu for some of them and it was working more or less fine, except collabora, which permanently drops black screens for any files - docx, xlsx and I can see FAILED in the console of the browser) 
* that's all, why I need to adjust the mtu for everything on the local network

Thank you!

**Comment by Zoey2936** [comment, marked as answer]
see here; http://www.microhowto.info/howto/persistently_change_the_mtu_of_a_network_interface_on_debian.html

**Comment by pongraczi** [comment]
Thank you @Zoey2936 for your reply!

I think I didn't describe my problem precisely enough, let me fix it.  

I have problem with the docker network settings.  
The LXC itself has correct mtu (1400), which needed to set in the config file of the lxc container by adding manually mtu=1400 and it is persistent between start-stop-restart-reboot.  

In fact, my setup looks like this:  

```
Servers                                 Request flow
Network          Internet          │
      ┌──────────Public IP         │    ┌─────────┐
      │          https             │    │ Browser │
┌─────┼─────────────────────────┐  │    │  https  │
│ WAN │      Firewall,          │  │    └────┬────┘
│   vmbr0────port fwd─────vmbr1 │  │         │
│            route          │   │  │    ┌────▼────┐
│                           │   │  │    │ProxmoxIP│
│                           │   │  │    │ prt.fwd │
│LAN───┬──────────────┬─────┴── │  │    └────┬────┘
│      │              │         │  │         │
│ ┌────┴────┐    ┌────┴────┐    │  │    ┌────▼────┐
│ │   eth0  │    │   eth0  │    │  │    │ Apache  │
│ │Apache   │    │Docker   │    │  │    │  http   │
│ │rev.proxy│    │NAIO     │    │  │    └────┬────┘
│ │10.x.x.5 │    │10.x.x.10│    │  │         │
│ │443      │    │11000    │    │  │    ┌────▼────┐
│ └─────────┘    └─────────┘    │  │    │ Docker  │
│                   Proxmox VE  │  │    │  NAIO   │
└───────────────────────────────┘  │    └─────────┘
```

I would like to see that, the docker containers inside the **Docker** lxc have mtu=1400.  
At this moment I set it via manually, when all of them are up, using the command inside the lxc: *ip link set group default mtu 1400*

**Reply by Zoey2936** [reply]
maybe use as workaround crontab `@reboot ip link show group default mtu 1400` ?

**Reply by Zoey2936** [reply]
but I think you should ask this question better in a docker forum or something like this

**Reply by pongraczi** [reply]
I assume, as the master container starts the remaining containers, there should be a location, where start sh files or config files exist, which are suitable for such a settings.

**Reply by Zoey2936** [reply]
if you need to set it with linux tools, then I think we can't make it work with docker options

**Reply by pongraczi** [reply]
Thank you, never mind, I reset my learning instance and started again to learn, how it works in my environment. I eliminated all the messy hacks I did today and I follow the right path.  
Maybe I do not really need it, at least up to now it is working as expected (can access to aio admin/login to nextcloud, browsing settings etc.).  
So, this discussion could be a hint for others: "changing mtu in docker not always necessary'.

**Comment by pongraczi** [comment]
SOLVED.  

Just for the record, in my case, I had to check and fix:  
* When the lxc CT has correct mtu (in my case 1400 is mandatory), based on my experience, docker settings doesn't matter, could be the default. (cloning resets the mtu to the default)  
* Fixing default route. (new environment and route was necessary)
* The firewall rules needed to set of the host server (to route back the 80/443 when destination is the own IP).
